pipeline/upi_test_executor.groovy

```diff
@@ -0,0 +1,167 @@
+// Workflow to execute tests
+def sharedLib
+def retVal
+def run_type = "RHCEPH-test-executor run"
+def rhcephVersion= "${params.RHCS_Build}"
+def phase = "${params.Build}"
+def group = "${params.Group}-"
+def suite = "${params.Suite}"
+def buildUserId
+def buildUserEmail
+def buildUserName
+
+def get_ceph_version(){
+    def cephVersion = 'quincy'
+    if(params.RHCS_Build.contains("5.")){
+        cephVersion = 'pacific'
+    }
+    else if(params.RHCS_Build.contains("3.")){
+        cephVersion = 'luminous'
+    }
+    else if(params.RHCS_Build.contains("4.")){
+        cephVersion = 'nautilus'
+    }
+    return cephVersion
+}
+
+def getCLI(){
+    /*
+        Generates the CLI using the arguments provided and returns it.
+    */
+    def cephVersion = 'quincy'
+    if(params.RHCS_Build.contains("5.")){
+        cephVersion = 'pacific'
+    }
+    else if(params.RHCS_Build.contains("3.")){
+        cephVersion = 'luminous'
+    }
+    else if(params.RHCS_Build.contains("4.")){
+        cephVersion = 'nautilus'
+    }
```

Comment:
Let's take out this as single definition in another groovy module. let's not duplicate same logic.